random1on1/api/config.py

```diff
@@ -1,22 +1,41 @@
 import json
 
+from preconditions import preconditions
+
+DEFAULT_ROLE = "Random-1-on-1s"
+DEFAULT_CATEGORY = "Random-1-on-1s"
+DEFAULT_HISTORY_CHANNEL = "pairing-history"
+DEFAULT_LOGGING_CHANNEL = "logs"
+DEFAULT_ANNOUNCEMENT_CHANNEL = "random-1-on-1-announcements"
+
 
 class Random1on1BotConfig(object):
-    # TODO: Add configuration values and precondition checks here
 
+    @preconditions(
+        lambda guild_id: len(guild_id) > 0,
+        lambda random1on1_role: len(random1on1_role) > 0,
+        lambda channel_category: len(channel_category) > 0,
+        lambda history_channel: len(history_channel) > 0,
+        lambda logging_channel: len(logging_channel) > 0,
+        lambda announcement_channel: len(announcement_channel) > 0,
+    )
```

Comment:
Seems like a lot of unecessary boiler plate code.

Would like to see a function and a test.

Reply:
e.g. `check_string_arg` and then use that in all the preconditions?

I can look into that, but might be restricted a bit by how the preconditions library does argument matching (happy to switch libraries, but would definitely want to generally keep preconditions as a separate thing from the body of each method).